code/modules/surgery/bodyparts/bodyparts.dm

```diff
@@ -73,16 +74,19 @@
 	if(ishuman(C))
 		var/mob/living/carbon/human/H = C
 		if(C.has_trait(TRAIT_LIMBATTACHMENT))
-			if(!H.get_bodypart(body_zone) && !animal_origin)
-				if(H == user)
-					H.visible_message("<span class='warning'>[H] jams [src] into [H.p_their()] empty socket!</span>",\
-					"<span class='notice'>You force [src] into your empty socket, and it locks into place!</span>")
-				else
-					H.visible_message("<span class='warning'>[user] jams [src] into [H]'s empty socket!</span>",\
-					"<span class='notice'>[user] forces [src] into your empty socket, and it locks into place!</span>")
-				user.temporarilyRemoveItemFromInventory(src, TRUE)
-				attach_limb(C)
-				return
+			if(((src.status == BODYPART_ORGANIC) && (!(ROBOTIC_LIMBS in H.dna.species.species_traits))) || ((src.status == BODYPART_ROBOTIC) && (ROBOTIC_LIMBS in H.dna.species.species_traits))) // Can't mix organic and robotic.
```

Comment:
i understand that a robot could not integrate a flesh arm, but i'd be sad if skeletons couldn't just attach terminator arms if they find any

Reply:
Fixed this for you spooky folks now, just need to commit the change once I finish the mob_biotype thing.